db/migrate/20200129172534_add_epoch_version_release_arch_to_katello_installed_packages.rb

```diff
@@ -0,0 +1,44 @@
+class AddEpochVersionReleaseArchToKatelloInstalledPackages < ActiveRecord::Migration[5.2]
+  def up
+    add_column :katello_installed_packages, :nvrea, :string
+    add_column :katello_installed_packages, :epoch, :string
+    add_column :katello_installed_packages, :version, :string
+    add_column :katello_installed_packages, :release, :string
+    add_column :katello_installed_packages, :arch, :string
+
+    epoch_non_0 = ::Katello::Rpm.where.not(epoch: 0)
+
+    ::Katello::InstalledPackage.reset_column_information
+    ::Katello::InstalledPackage.all.each do |pkg|
+      epoch = epoch_non_0.where(nvra: pkg.nvra).first
+      if epoch.nil?
+        epoch = '0'
+      end
+
+      pkg.epoch = epoch
+
+      nvra = pkg.nvra
+      arch = nvra[nvra.rindex('.') + 1..-1]
+      nvra.sub('.' + arch, '')
+
+      tokens = nvra.split('-')
+      release = tokens[-1]
+      version = tokens[-2]
+
+      pkg.arch = arch
+      pkg.release = release
+      pkg.version = version
+
+      pkg.nvrea = "#{epoch}:#{pkg.name}-#{version}-#{release}.#{arch}"
+      pkg.save!
+    end
+  end
```

Comment:
Should we set the epoch to zero for all the other installed packages after this runs?